### ISSUE TITLE
diag and note messages don't show up in prove output

### ISSUE THREAD

**Issue by szabgab** [pherkin/test-bdd-cucumber-perl#180]
I called 'diag' and 'note' in the *_steps.pl file, but they are not displayed when I run `prove`.
Not even with 'prove -v`

I think it would be very useful to be able to see these messages, or to have some other way to add comments to the steps scripts that will show up in the output of `prove`

**Comment by ehuelsmann** [comment]
This behaviour can be explained from the fact that the test results (and `note` and `diag` output) is *not* used to generate output for `prove` to be consumed. Instead, the output is captured to determine test-success and from that, feed the formatters [one of which is a TAP formatter, which generates a new test result for each Given/When/Then line executed].

**Comment by ehuelsmann** [comment]
hmm. I guess one of the questions could be: what does`prove` do with notes and diags inside subtests?

**Comment by ehuelsmann** [comment]
Ok. The behaviour that I'm seeing is to print *either* the diag output *or* the note output (but not both), even for subtests, even when `prove` does not run in verbose mode (although without verbose, the notes are not printed; when verbose mode chooses the note, no output appears for the non-verbose mode).

**Comment by DragosTrif** [comment]
When I played around with this i noticed this behavior: When all the test pass note and diag are not printed.  If the test fails both are printed.
The command i used to test this is:
`carton exec -- prove -rv -m --source Feature --feature-option exetension=.feature --ext=.feature --source=Perl --ext=.t nesting`
The code run by this is:
``
The ouptut:
`# Failed test 'Given input of Fred Bloggs'
# at nesting/features/nesting.feature line 7.
#   in step at nesting/features/nesting.feature line 7.
#     # test note
#     # test diag 1
#     not ok 2 - test
#     #   Failed test 'test'
#     #   at /home/dragos/projects/cucumber/nesting/features/step_definitions/nesting_steps.pl line 34.
#     #          got: '1'
#     #     expected: '0'
#     # test diag 2
#     ok 6 - Step ran to completion
#     1..2
#     # Looks like you failed 1 test of 2.`

**Comment by ehuelsmann** [comment]
that's correct: on error *all* TAP output of the step is shown. The TAP output gets captured by the Test::BDD::Cucumber harness and needs to be relayed in some form or another to `prove`. Currently it only does that on failure. On success, *no* output of  the TAP is shown, because it never enters the TAP stream `prove` consumes: It's been eaten by the Test::BDD::Cucumber::Harness.

**Comment by ehuelsmann** [comment]
I'm wondering how to show the diag output even if the test did not fail.
